src/main/java/com/radixdlt/crypto/ECKeyPair.java

```diff
@@ -161,4 +192,69 @@ public String toString() {
 		return String.format("%s[%s]",
 			getClass().getSimpleName(), Bytes.toBase64String(getPublicKey().getBytes()));
 	}
+
+	// ###  From Client Library  ###
+
+	@JsonProperty("version")
+	@DsonOutput(DsonOutput.Output.ALL)
+	private short version = 100;
+
+	// Placeholder for the serializer ID
+	@JsonProperty(SerializerConstants.SERIALIZER_NAME)
+	@DsonOutput(DsonOutput.Output.ALL)
+	private SerializerDummy serializer = SerializerDummy.DUMMY;
+
+	public ECKeyPair(byte[] publicKey, byte[] privateKey) {
+		try {
+			this.publicKey = new ECPublicKey(publicKey);
+		} catch (CryptoException e) {
+			throw new IllegalArgumentException("Failed to create public key from bytes, error: " + e);
+		}
+		this.privateKey = Arrays.copyOf(privateKey, privateKey.length);
+	}
+
+	public static ECKeyPair fromFile(File file) throws CryptoException {
+		try (BufferedInputStream io = new BufferedInputStream(new FileInputStream(file))) {
+			byte[] privateKey = new byte[32];
+			int len = io.read(privateKey, 0, 32);
+
+			if (len < 32) {
+				throw new IllegalStateException("Private Key file must be 32 bytes");
```

Comment:
Exception thrown here but caught a few lines later in the catch-all.

Reply:
yeah most code here is just copy paste over, some of your comments will not be addressed, especially code related to crypto, because in coming PR I will make some changes and improvements to the crypto related code. See [RPNV1-62](https://radixdlt.atlassian.net/browse/RPNV1-62).

Reply:
I changed my mind! Will address these things now, but might be change later anyway. Thanks for pointing this out, improving this.